### DISCUSSION TITLE
Issue with configuration files

### DISCUSSION THREAD

**Discussion by HaocongYang1** [PlusToolkit/PlusLib#1003] [Q&A]
Hi, I currently have the Ascension Trackstar, Siemens P300 ultrasound, and the Epiphan frame grabber VGA2USB, and I am trying to link all three hardware through PLUS to Slicer. Individually, through the 'Epiphan video capture device' and 'Ascension 3DG tracker' configurations, I am able to make the sensor probes and captures ultrasound screen appear through OpenIGT on Slicer. But, I cannot make them appear together in Slicer since I can only connect to 1 server/configuration at a time. 

So, in this case, do I need to write a configuration file for my specific setup that combines the two devices? How can I write such a configuration and what level of programming do I need to be at? I tried pasting the Epiphan file commands into the Ascension config. but it did not work. 

Are there other options of connection other than writing a completely new configuration? Appreciate the time!

**Comment by adamrankin** [comment]
If you want a single server to provide both images and tracking, you'll have to combine them using a mixer device. See this config file for an example:

https://github.com/PlusToolkit/PlusLibData/blob/master/ConfigFiles/PlusDeviceSet_Server_Philips_ie33_NDIAurora.xml

**Comment by HaocongYang1** [comment]
Hi, thanks for the answer and sorry for the delayed response. So when you say mixer device does that mean simply a config file that mixes/combines the two devices signals or an additional physical mixer device? From looking at the config file I assume it is the former?

Also, what programming language / command type is the PLUS config files written in? Is it the XML markup language? Are there any resource or tutorials that you would suggest on writing config files? Thanks!

**Comment by adamrankin** [comment]
It is indeed the former. It is a software devices that combines channels.

The config files are in XML, and unfortunately there is no guide for the config files. We generally construct them from example, or by referencing the device documentation here:

http://perk-software.cs.queensu.ca/plus/doc/nightly/user/Configuration.html

**Reply by lassoan** [reply]
Pages under http://perk-software.cs.queensu.ca/plus/doc/nightly/user/Configuration.html is the guide. You can find detailed specification of all XML elements and attributes and lots of examples. If anything is missing or unclear then let us know. There are also step-by-step tutorials at:
- https://github.com/PerkLab/PerkLabBootcamp/tree/master/Doc and
- http://www.slicerigt.org/wp/user-tutorial/
